src-v2/Judge/NTS.Judge/Adapters/Behinds/StartlistBehind.cs

```diff
@@ -9,37 +9,25 @@
 
 namespace NTS.Judge.Adapters.Behinds;
 
-public class StartlistBehind : ObservableBehind, IStartupInitializer, IStartlistBehind
+public class StartlistBehind : ObservableBehind, IStartlistUpcoming, IStartlistHistory
 {
     readonly IRepository<Participation> _participationRepository;
 
     public StartlistBehind(IRepository<Participation> participationRepository)
     {
         _participationRepository = participationRepository;
-        var startlist = new StartList();
+        var startlist = new StartList(EmitChange);
         Startlist = startlist;
     }
 
-    public StartList Startlist { get; set; }
-    public List<Start> Upcoming => Startlist.Upcoming;
-    public List<Start> History => Startlist.History;
+    StartList Startlist { get; set; }
+    public IReadOnlyList<Start> Upcoming => Startlist.Upcoming;
+    public IReadOnlyList<Start> History => Startlist.History;
 
     protected override async Task<bool> PerformInitialization(params IEnumerable<object> arguments)
     {
         var participations = await _participationRepository.ReadAll();
         Startlist.AssignStarts(participations);
```

Comment:
Remove the AssignStarts but as consequence had to ad the following logic:
```
StartList? _startlist;
...
 public IReadOnlyList<Start> Upcoming => _startlist?.Upcoming == null ? [] : _startlist.Upcoming;
 public IReadOnlyList<Start> History => _startlist?.History == null? [] : _startlist.History;
 ```
 to avoid the error **Non-nullable field must contain a non-null value when exiting constructor. Consider declaring it as nullable.** in the ctor.

Reply:
That's right. You can do a bit better however:
```c#
public IReadOnlyList<Start> Upcoming => _startlist?.Upcoming ?? [];
```

And if you think about it it makes more sense, because you previously had an empty Startlist. What happens if you omitted to assign default values to `History` or `Upcoming`? This way is safer. That's a good illustration of the usefulness of Nullable